core/src/main/java/com/datastax/oss/driver/internal/core/data/DefaultTupleValue.java

```diff
@@ -93,6 +95,44 @@ private void readObject(ObjectInputStream stream) throws InvalidObjectException
     throw new InvalidObjectException("Proxy required");
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+
+    if (o == null || !(o instanceof TupleValue)) return false;
```

Comment:
The `instanceof` check fails if o is null, so you don't need the explicit null check.